frictionless/report/report.py

```diff
@@ -166,6 +170,119 @@ def wrapper(*args, **kwargs):
+            # for zipped resources append file name
+            if task.resource.innerpath:
+                source = f"{source} => {task.resource.innerpath}"
+            validation_content += f"\n# {'-'*len(prefix)}"
+            validation_content += f"\n# {prefix}: {source} {suffix}"
+            validation_content += f"\n# {'-'*len(prefix)}"
+            error_list = {}
+            error_content = []
+            if task.errors:
+                for error in task.errors:
+                    error_content.append(
+                        [
+                            error.get("rowPosition", ""),
+                            error.get("fieldPosition", ""),
+                            error.code,
+                            error.message,
+                        ]
+                    )
+                    # error list for summary
+                    error_title = f"{error.name} ({error.code})"
+                    if error_title not in error_list:
+                        error_list[error_title] = 0
+                    error_list[error_title] += 1
+                    if task.partial:
+                        last_row_checked = error.get("rowPosition", "")
+            # Validate
+            error_content = _wrap_text_to_colwidths(error_content)
+            rows_checked = last_row_checked if task.partial else None
+            summary_content = self.validation_summary(
+                source,
+                basepath=task.resource.basepath,
+                time_taken=self.time,
+                rows_checked=rows_checked,
+                error_list=error_list,
+            )
+            validation_content += "\n\n"
+            validation_content += "## Summary "
+            validation_content += "\n\n"
+            if task.partial:
+                validation_content += "\n\n"
+                validation_content += (
+                    "The document was partially validated because of one of the limits\n"
+                )
+                validation_content += "* limit errors"
+                validation_content += "* memory Limit"
+                validation_content += "\n\n"
+            validation_content += str(
+                tabulate(
+                    summary_content,
+                    headers=["Description", "Size/Name/Count"],
+                    tablefmt="grid",
+                )
+            )
+            validation_content += "\n\n"
+            # errors
+            if task.errors:
+                validation_content += "## Errors "
+                validation_content += "\n\n"
+                validation_content += str(
+                    tabulate(
+                        error_content,
+                        headers=["row", "field", "code", "message"],
+                        tablefmt="grid",
+                    )
+                )
+                validation_content += "\n\n"
+
+        return validation_content
+
+    def validation_summary(
```

Comment:
@shashigharti 
I think the last step will be moving this method to be `reportTask.to_summary` so you don't need to pass any arguments? The description says "Generate summary for validation task" which is a clear indicator that "self" in this method should be a `reportTask` object I guess